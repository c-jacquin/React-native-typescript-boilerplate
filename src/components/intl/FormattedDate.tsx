import React, { StatelessComponent } from 'react'
import { FormattedDate } from 'react-intl'
import { Text } from 'react-native'
import { IFormattedDateProps } from './types'

const NativeFormattedDate: StatelessComponent<IFormattedDateProps> = props =>
    <FormattedDate {...props}>
        {(localized: string) =>
            <Text style={props.style}>
                {localized}
            </Text>}
    </FormattedDate>

export default NativeFormattedDate
