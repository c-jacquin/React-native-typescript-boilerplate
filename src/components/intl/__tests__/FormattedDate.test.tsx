import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'

import FormattedDate from '../FormattedDate'

describe('FormattedDate', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <IntlProvider locale="en">
                    <FormattedDate style={{}} value={1500} />
                </IntlProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
