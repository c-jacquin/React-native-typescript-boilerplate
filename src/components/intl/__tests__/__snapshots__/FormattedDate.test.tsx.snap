// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedDate should renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  1/1/1970
</Text>
`;
